File: backend/schemas/body_metric.py
from pydantic import BaseModel, ConfigDict, Field
from datetime import date

# Base schema for body metric.
class BodyMetricBase(BaseModel):
    date: date
    body_weight: float = Field(..., gt=0)

# Schema for creating a new body metric entry.
class BodyMetricCreate(BodyMetricBase):
    pass

# Schema for the output sent back to the client.
class BodyMetric(BodyMetricBase):
    id: int
    user_id: int

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/schemas/body-metric.ts
import { z } from 'zod'

export const BodyMetricSchema = z.object({
  id: z.number(),
  user_id: z.number(),
  date: z.iso.date(),
  body_weight: z.number(),
})
export type BodyMetric = z.infer<typeof BodyMetricSchema>

export const BodyMetricCreateSchema = z.object({
  date: z.iso.date(),
  body_weight: z.number().gt(0),
})
export type BodyMetricCreate = z.infer<typeof BodyMetricCreateSchema>
